scripts/actions/send-and-update-translation-queue.js

```diff
@@ -0,0 +1,200 @@
+const fs = require('fs');
+const path = require('path');
+const FormData = require('form-data');
+const fetch = require('node-fetch');
+
+const loadFromDB = require('./utils/load-from-db');
+const saveToDB = require('./utils/save-to-db');
+const serializeMDX = require('./serialize-mdx');
+const { vendorRequest, getAccessToken } = require('./utils/vendor-request');
+
+/**
+ * @typedef {Object} Page
+ * @property {string} file The filepath for the page (from the project root).
+ * @property {string} html The HTML serialized content for the page.
+ */
+
+// NOTE: the vendor requires the locales in a different format
+// We should consider this into the Gatsby config for each locale.
+const LOCALE_IDS = {
+  jp: 'ja-JP',
+};
+
+const PROJECT_ID = process.env.TRANSLATION_VENDOR_PROJECT;
+
+/**
+ * Take a list of filepaths (grouped by locale) and fetches the HTML content.
+ * NOTE: async inside of loops is hard.
+ * @param {Object<string, string[]>} locales The queue of slugs to be translated.
+ * @returns {Promise<Object<string, Page[]>>}
+ */
+const getContent = async (locales) =>
+  Object.entries(locales).reduce(async (accP, [locale, slugs]) => {
+    const acc = await accP;
```

Comment:
I'd love some input on this function. We're using `reduce` with async functions. The accumulator isn't the object, but rather a _promise_. I thought we needed to await it to ensure that we could correctly spread on it (below).

Reply:
> The accumulator isn't the object, but rather a _promise_. 

I see that the object _values_ are promises, but the [accumulator itself looks like an object](The accumulator isn't the object, but rather a promise. ) to me. Am I missing something?

Reply:
Oh wait, is this because the reduce function is an `async` function that resolves to an object?

Reply:
Yeah...we've ran into this situation in the past. Because the reduce callback function is `async`, the return value is a promise. Which means that the first parameter (the accumulator) is a promise, not the value we want.

There's some serious code smell funk going on with this function, but I'm not sure how to break this up / synchronize it.

Reply:
Rather than `await`ing the value of the promise, you can do that later on to clean this up, which makes this just a normal reduce function.

My suggestion would be to remove the `async`/`await` here and instead amend [this code](https://github.com/newrelic/docs-website/pull/581/files#diff-3275c521907ff8a8e4237b2b80241b357a478e451ec141a3c21300fdfcbb7df0R120) to something like the following:

```js
const pages = await Promise.all(Object.values(content));
```